### ISSUE TITLE
gsutil isn't authenticated using a service account key

### ISSUE THREAD

**Issue by jamiet-msm** [google-github-actions/auth#129]
### TL;DR

gsutil is not authenticated when using auth action

Not fair to call this a bug, I suspect user error is involved somewhere, but Seth Vargo suggested I post this question here (see [tweet](https://twitter.com/sethvargo/status/1489240360299765768?s=20&t=RMs9kF10utvPp_NY9Oh4ZA))

If someone can explain how to accomplish this I'd be very grateful.

### Expected behavior

using google-github-actions/auth would provide authentication for gsutil

### Observed behavior

gcloud is authenticated but gsutil is not.

The provided action yaml contains 2 jobs. Both of them run a gcloud command and a gsutil command. One of them authenticates using the old mechanism using google-github-actions/setup-gcloud, the other authenticates using google-github-actions/auth.

The one that authenticates using google-github-actions/setup-gcloud works
![image](https://user-images.githubusercontent.com/79862277/152362681-88e46d5f-3111-48ef-aac9-2d75f6daad47.png)

The one that authenticates using google-github-actions/auth fails on the gsutil command with error 

> Anonymous caller does not have storage.objects.list access to the Google Cloud Storage bucket

![image](https://user-images.githubusercontent.com/79862277/152362566-93cd970b-8f7f-4b89-b516-4f5fa1be2129.png)

### Action YAML

```yaml
name: demo-auth-issue

on:
  push:
  workflow_dispatch:

jobs:
  auth-using-auth:
    runs-on: ubuntu-latest
    steps:
      - id: auth
        uses: google-github-actions/auth@v0.4.0
        with:
          credentials_json: ${{ secrets.SA_JSON_KEY }}
      - name: Set up Cloud SDK
        uses: google-github-actions/setup-gcloud@v0
      - name: Install gcloud alpha
        shell: bash
        run: |
          gcloud components install alpha --quiet
      - name: 'Use gcloud CLI'
        run: gcloud alpha storage ls gs://dev-dp-braze-payloads
      - name: run gsutil
        shell: bash
        run: |
          gsutil du -sh gs://dev-dp-braze-payloads

  auth-using-setup-gcloud:
    runs-on: ubuntu-latest
    steps:
      - id: auth
        uses: google-github-actions/setup-gcloud@v0
        with:
          service_account_key: ${{ secrets.SA_JSON_KEY }}
      - name: Install gcloud alpha
        shell: bash
        run: |
          gcloud components install alpha --quiet
      - name: 'Use gcloud CLI'
        run: gcloud alpha storage ls gs://dev-dp-braze-payloads
      - name: run gsutil
        shell: bash
        run: |
          gsutil du -sh gs://dev-dp-braze-payloads
```

### Log output

```text
2022-02-03T14:05:04.4964520Z Waiting for a runner to pick up this job...
2022-02-03T14:05:05.2490074Z Job is waiting for a hosted runner to come online.
2022-02-03T14:05:11.4659947Z Job is about to start running on the hosted runner: Hosted Agent (hosted)
2022-02-03T14:05:14.9432200Z Current runner version: '2.287.1'
2022-02-03T14:05:14.9468079Z ##[group]Operating System
2022-02-03T14:05:14.9468980Z Ubuntu
2022-02-03T14:05:14.9469621Z 20.04.3
2022-02-03T14:05:14.9470116Z LTS
2022-02-03T14:05:14.9470647Z ##[endgroup]
2022-02-03T14:05:14.9471822Z ##[group]Virtual Environment
2022-02-03T14:05:14.9472517Z Environment: ubuntu-20.04
2022-02-03T14:05:14.9473223Z Version: 20220131.1
2022-02-03T14:05:14.9474041Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220131.1/images/linux/Ubuntu2004-Readme.md
2022-02-03T14:05:14.9475054Z Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220131.1
2022-02-03T14:05:14.9475858Z ##[endgroup]
2022-02-03T14:05:14.9476452Z ##[group]Virtual Environment Provisioner
2022-02-03T14:05:14.9477088Z 1.0.0.0-main-20220128-1
2022-02-03T14:05:14.9477749Z ##[endgroup]
2022-02-03T14:05:14.9479639Z ##[group]GITHUB_TOKEN Permissions
2022-02-03T14:05:14.9480779Z Actions: write
2022-02-03T14:05:14.9481602Z Checks: write
2022-02-03T14:05:14.9482270Z Contents: write
2022-02-03T14:05:14.9482839Z Deployments: write
2022-02-03T14:05:14.9483398Z Discussions: write
2022-02-03T14:05:14.9484094Z Issues: write
2022-02-03T14:05:14.9484640Z Metadata: read
2022-02-03T14:05:14.9485215Z Packages: write
2022-02-03T14:05:14.9485769Z Pages: write
2022-02-03T14:05:14.9486336Z PullRequests: write
2022-02-03T14:05:14.9486997Z RepositoryProjects: write
2022-02-03T14:05:14.9487702Z SecurityEvents: write
2022-02-03T14:05:14.9488255Z Statuses: write
2022-02-03T14:05:14.9488964Z ##[endgroup]
2022-02-03T14:05:14.9493961Z Secret source: Actions
2022-02-03T14:05:14.9494739Z Prepare workflow directory
2022-02-03T14:05:15.0633087Z Prepare all required actions
2022-02-03T14:05:15.0890430Z Getting action download info
2022-02-03T14:05:15.3135922Z Download action repository 'google-github-actions/auth@v0.4.0' (SHA:a61909d048e0be579b6c15b27088d19668493851)
2022-02-03T14:05:15.6628847Z Download action repository 'google-github-actions/setup-gcloud@v0' (SHA:e0f83f24f541c30f1e9fe3c966963a10b8e647f5)
2022-02-03T14:05:16.1005374Z ##[group]Run google-github-actions/auth@v0.4.0
2022-02-03T14:05:16.1006150Z with:
2022-02-03T14:05:16.1014569Z   credentials_json: ***

2022-02-03T14:05:16.1015245Z   create_credentials_file: true
2022-02-03T14:05:16.1015907Z   access_token_lifetime: 3600s
2022-02-03T14:05:16.1016667Z   access_token_scopes: https://www.googleapis.com/auth/cloud-platform
2022-02-03T14:05:16.1017393Z   id_token_include_email: false
2022-02-03T14:05:16.1017962Z ##[endgroup]
2022-02-03T14:05:16.3316185Z 
2022-02-03T14:05:16.3390601Z 
2022-02-03T14:05:16.3581713Z ##[group]Run google-github-actions/setup-gcloud@v0
2022-02-03T14:05:16.3582410Z with:
2022-02-03T14:05:16.3582832Z   version: latest
2022-02-03T14:05:16.3583325Z   export_default_credentials: false
2022-02-03T14:05:16.3583863Z   cleanup_credentials: true
2022-02-03T14:05:16.3584334Z env:
2022-02-03T14:05:16.3584928Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:16.3585702Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:16.3586409Z   CLOUDSDK_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:16.3587083Z   CLOUDSDK_CORE_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:16.3587742Z   GCP_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:16.3588391Z   GCLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:16.3589048Z   GOOGLE_CLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:16.3589625Z ##[endgroup]
2022-02-03T14:05:17.8140059Z [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/af936b24-50c8-4b0b-8703-126bd1db6dec -f /home/runner/work/_temp/30c3fc13-a60c-4add-81f0-d00f3b894f80
2022-02-03T14:05:28.9180485Z No credentials detected, skipping authentication
2022-02-03T14:05:28.9404651Z ##[group]Run gcloud components install alpha --quiet
2022-02-03T14:05:28.9405309Z [36;1mgcloud components install alpha --quiet[0m
2022-02-03T14:05:28.9480074Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail ***0***
2022-02-03T14:05:28.9480580Z env:
2022-02-03T14:05:28.9481060Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:28.9481687Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:28.9482385Z   CLOUDSDK_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:28.9482890Z   CLOUDSDK_CORE_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:28.9483526Z   GCP_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:28.9484144Z   GCLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:28.9484783Z   GOOGLE_CLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:28.9485456Z   CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
2022-02-03T14:05:28.9486093Z ##[endgroup]
2022-02-03T14:05:31.0634545Z 
2022-02-03T14:05:31.0640518Z 
2022-02-03T14:05:31.0641335Z Your current Cloud SDK version is: 371.0.0
2022-02-03T14:05:31.0642590Z Installing components from version: 371.0.0
2022-02-03T14:05:31.0643537Z 
2022-02-03T14:05:31.0673968Z ┌──────────────────────────────────────────────┐
2022-02-03T14:05:31.0676778Z │     These components will be installed.      │
2022-02-03T14:05:31.0679265Z ├───────────────────────┬────────────┬─────────┤
2022-02-03T14:05:31.0681438Z │          Name         │  Version   │   Size  │
2022-02-03T14:05:31.0683297Z ├───────────────────────┼────────────┼─────────┤
2022-02-03T14:05:31.0695992Z │ gcloud Alpha Commands │ 2022.01.28 │ < 1 MiB │
2022-02-03T14:05:31.0697590Z └───────────────────────┴────────────┴─────────┘
2022-02-03T14:05:31.0698543Z 
2022-02-03T14:05:31.3006678Z For the latest full release notes, please visit:
2022-02-03T14:05:31.3007686Z   https://cloud.google.com/sdk/release_notes
2022-02-03T14:05:31.3008207Z 
2022-02-03T14:05:31.3071781Z ╔════════════════════════════════════════════════════════════╗
2022-02-03T14:05:31.3073227Z ╠═ Creating update staging area                             ═╣
2022-02-03T14:05:34.9390965Z ╚════════════════════════════════════════════════════════════╝
2022-02-03T14:05:35.2212766Z ╔════════════════════════════════════════════════════════════╗
2022-02-03T14:05:35.2213854Z ╠═ Installing: gcloud Alpha Commands                        ═╣
2022-02-03T14:05:35.3278311Z ╚════════════════════════════════════════════════════════════╝
2022-02-03T14:05:35.3422977Z ╔════════════════════════════════════════════════════════════╗
2022-02-03T14:05:35.3424293Z ╠═ Creating backup and activating new installation          ═╣
2022-02-03T14:05:35.3425456Z ╚════════════════════════════════════════════════════════════╝
2022-02-03T14:05:35.3446821Z 
2022-02-03T14:05:55.1362650Z Performing post processing steps...
2022-02-03T14:05:55.1363493Z ..............................................................................done.
2022-02-03T14:05:55.1453454Z 
2022-02-03T14:05:55.1453764Z Update done!
2022-02-03T14:05:55.1454027Z 
2022-02-03T14:05:55.1454305Z WARNING:   There are other instances of Google Cloud tools on your system PATH.
2022-02-03T14:05:55.1455113Z   Please remove the following to avoid confusion or accidental invocation:
2022-02-03T14:05:55.1455436Z 
2022-02-03T14:05:55.1455932Z   /usr/lib/google-cloud-sdk/bin/gsutil
2022-02-03T14:05:55.1456436Z /usr/lib/google-cloud-sdk/bin/anthoscli
2022-02-03T14:05:55.1457009Z /usr/lib/google-cloud-sdk/bin/docker-credential-gcloud
2022-02-03T14:05:55.1457537Z /usr/lib/google-cloud-sdk/bin/bq
2022-02-03T14:05:55.1458080Z /usr/lib/google-cloud-sdk/bin/git-credential-gcloud.sh
2022-02-03T14:05:55.1458610Z /usr/lib/google-cloud-sdk/bin/gcloud
2022-02-03T14:05:55.1458850Z 
2022-02-03T14:05:55.1458971Z   
2022-02-03T14:05:55.2692841Z ##[group]Run gcloud alpha storage ls gs://dev-dp-braze-payloads
2022-02-03T14:05:55.2693490Z [36;1mgcloud alpha storage ls gs://dev-dp-braze-payloads[0m
2022-02-03T14:05:55.2758198Z shell: /usr/bin/bash -e ***0***
2022-02-03T14:05:55.2758748Z env:
2022-02-03T14:05:55.2759189Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:55.2759765Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:55.2760444Z   CLOUDSDK_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:55.2761131Z   CLOUDSDK_CORE_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:55.2761966Z   GCP_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:55.2762446Z   GCLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:55.2762940Z   GOOGLE_CLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:55.2763452Z   CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
2022-02-03T14:05:55.2763940Z ##[endgroup]
2022-02-03T14:05:56.5288652Z ##[group]Run gsutil du -sh gs://dev-dp-braze-payloads
2022-02-03T14:05:56.5289243Z [36;1mgsutil du -sh gs://dev-dp-braze-payloads[0m
2022-02-03T14:05:56.5352449Z shell: /usr/bin/bash --noprofile --norc -e -o pipefail ***0***
2022-02-03T14:05:56.5353402Z env:
2022-02-03T14:05:56.5354381Z   CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:56.5355039Z   GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/_temp/4e13689b43732933b32cb83c
2022-02-03T14:05:56.5355640Z   CLOUDSDK_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:56.5356273Z   CLOUDSDK_CORE_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:56.5356773Z   GCP_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:56.5357259Z   GCLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:56.5357791Z   GOOGLE_CLOUD_PROJECT: msm-groupdata-admin-data-eng
2022-02-03T14:05:56.5358321Z   CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
2022-02-03T14:05:56.5359030Z ##[endgroup]
2022-02-03T14:05:58.1419737Z ServiceException: 401 Anonymous caller does not have storage.objects.list access to the Google Cloud Storage bucket.
2022-02-03T14:05:58.3202905Z ##[error]Process completed with exit code 1.
2022-02-03T14:05:58.3269960Z Post job cleanup.
2022-02-03T14:05:58.3798496Z Skipping credential cleanup - "export_default_credentials" is false.
2022-02-03T14:05:58.4028518Z Cleaning up orphan processes
```

### Additional information

don't think so.

**Comment by sethvargo** [comment]
Can you please try pinning `auth` to @v0? There have been many changes since v0.4.0.
